spec/service_consumers/pact_helper.rb

```diff
@@ -106,6 +106,53 @@ def url_encode(str)
     end
   end
 
+  provider_state "there is a valid user session, with saved pages" do
+    set_up do
+      SavedPage.destroy_all
+      user = OidcUser.find_or_create_by(sub: "user-id")
+      FactoryBot.create_list(:saved_page, 2, oidc_user_id: user.id)
+    end
+
+    tear_down do
+      OidcUser.find_by(sub: "user-id").destroy!
+      SavedPage.destroy_all
+    end
+  end
+
+  provider_state "there is a valid user session, with /guidance/some-govuk-guidance saved" do
+    set_up do
+      user = OidcUser.find_or_create_by!(sub: "user-id")
+      FactoryBot.create(:saved_page, page_path: "/guidance/some-govuk-guidance", oidc_user_id: user.id)
+    end
+
+    tear_down do
+      OidcUser.find_by(sub: "user-id").destroy!
+      SavedPage.destroy_all
+    end
+  end
+
+  provider_state "there is a valid user session, a request to save /guidance/some-govuk-guidance saved" do
+    set_up do
+      OidcUser.find_or_create_by!(sub: "user-id")
+    end
+
+    tear_down do
+      OidcUser.find_by(sub: "user-id").destroy!
+    end
+  end
```

Comment:
Why is this needed over just `there is a valid user session`?

Reply:
humm there might be a clash no?
I could put either delete or save into the "there is a valid session",
But the happy path for each depends on opposite state (delete is happy if there was something to delete, create is happy if there was nothing to create?)

Reply:
A clash of what?  Right now this `provider_state` isn't setting anything up relating to saved pages